src/parser.rs

```diff
@@ -1201,23 +1201,32 @@ impl<'a> Parser<'a> {
                 negated,
             });
         }
-        self.expect_token(&Token::LParen)?;
-        let in_op = if self.parse_keyword(Keyword::SELECT) || self.parse_keyword(Keyword::WITH) {
-            self.prev_token();
-            Expr::InSubquery {
-                expr: Box::new(expr),
-                subquery: Box::new(self.parse_query()?),
-                negated,
-            }
+        if self.peek_token() == Token::LParen {
```

Comment:
As written, this code will now parse `<expr> IN <expr>` *anywhere* in the SQL query, not just in the position function which I don't think is correct

I wrote a test for the expected behavior in https://github.com/sqlparser-rs/sqlparser-rs/pull/468 (and that test does not pass with this branch)

Given that I finally see that `POSITION` does probably need to be special cased as only within `POSITION` would `<expr> IN <expr>` parse successfully.